pkg/subctl/cmd/gather/gather.go

```diff
@@ -126,44 +125,32 @@ func gatherData(cluster *cmd.Cluster) bool {
 func gatherDataByCluster(cluster *cmd.Cluster, directory string) {
 	var err error
 	clusterName := cluster.Name
+	status := cli.NewReporter()
 
-	fmt.Printf("Gathering information from cluster %q\n", clusterName)
+	clientProducer, err := client.NewProducerFromRestConfig(cluster.Config)
+	if err != nil {
+		status.Failure("Error creating client producer")
+
+		return
+	}
 
 	info := Info{
 		RestConfig:           cluster.Config,
 		ClusterName:          clusterName,
 		DirName:              directory,
 		IncludeSensitiveData: includeSensitiveData,
 		Summary:              &Summary{},
+		ClientProducer:       clientProducer,
+		Submariner:           cluster.Submariner,
 	}
 
-	info.DynClient = cluster.DynClient
-	info.ClientSet = cluster.KubeClient
-
-	submarinerClient, err := subOperatorClientset.NewForConfig(cluster.Config)
-	if err != nil {
-		fmt.Printf("Error getting Submariner client: %s\n", err)
-		return
-	}
-
-	info.Submariner, err = submarinerClient.SubmarinerV1alpha1().Submariners(cmd.OperatorNamespace).
-		Get(context.TODO(), submarinercr.SubmarinerName, metav1.GetOptions{})
-	if err != nil {
-		info.Submariner = nil
-
-		if !apierrors.IsNotFound(err) {
-			fmt.Printf("Error getting Submariner resource: %s\n", err)
-			return
-		}
-	}
-
-	info.ServiceDiscovery, err = submarinerClient.SubmarinerV1alpha1().ServiceDiscoveries(cmd.OperatorNamespace).
+	info.ServiceDiscovery, err = clientProducer.ForOperator().SubmarinerV1alpha1().ServiceDiscoveries(cmd.OperatorNamespace).
 		Get(context.TODO(), names.ServiceDiscoveryCrName, metav1.GetOptions{})
 	if err != nil {
 		info.ServiceDiscovery = nil
 
 		if !apierrors.IsNotFound(err) {
-			fmt.Printf("Error getting ServiceDiscovery resource: %s\n", err)
+			info.Status.Failure("Error getting ServiceDiscovery resource: %s", err)
```

Comment:
We haven't created `info.Status` yet.

```suggestion
			status.Failure("Error getting ServiceDiscovery resource: %s", err)
```